port_ocean/core/integrations/mixins/live_events.py

```diff
@@ -0,0 +1,156 @@
+            )
+        )
+
+    async def _get_live_event_resources(self, kind: str) -> list[ResourceConfig]:
+        try:
+            app_config = event.port_app_config
+            logger.info(
+                f"process data will use the following mappings: {app_config.dict()}"
+            )
+
+            resource_mappings = [
+                resource for resource in app_config.resources if resource.kind == kind
+            ]
+
+            if not resource_mappings:
+                logger.warning(f"No resource mappings found for kind: {kind}")
+                return []
+
+            return resource_mappings
+        except Exception as e:
+            logger.error(f"Error getting live event resources: {str(e)}")
+            raise
+
+    async def _export(self, resource_mapping: ResourceConfig, raw_item: RAW_ITEM) -> tuple[bool, list[str]]:
+        """Export a single resource mapping with the given raw item.
+
+        Args:
+            resource_mapping: The resource configuration to use for processing
+            raw_item: The raw data item to process
+
+        Returns:
+            tuple containing:
+                - bool: Whether the export succeeded
+                - list[str]: List of exported entity identifiers
+        """
+        try:
+            calculation_results = await self.entity_processor.parse_items(
+                resource_mapping, [raw_item], parse_all=True, send_raw_data_examples_amount=0
+            )
+
+            if not calculation_results or len(calculation_results.entity_selector_diff.passed)== 0:
+                logger.info(f"No entities passed selector for resource: {resource_mapping.kind}")
+                return True, []
+
+            upsertedEntities = await self.entities_state_applier.upsert(calculation_results.    entity_selector_diff.passed, UserAgentType.exporter)
+            return True, upsertedEntities
+
+        except Exception as e:
+            logger.error(f"Error exporting resource {resource_mapping.kind}: {str(e)}")
+            return False, []
+
+    async def _is_entity_exists(self, entity: Entity) -> bool:
+        """Check if this integration is the owner of the given entity.
+
+        Args:
+            entity: The entity to check ownership for
+
+        Returns:
+            bool: True if this integration is the owner of the entity, False otherwise
+        """
+        query = {
+            "combinator": "and",
+            "rules": [
+                {
+                    "property": "$identifier",
+                    "operator": "=",
+                    "value": entity.identifier
+                },
+                {
+                    "property": "$blueprint",
+                    "operator": "=",
+                    "value": entity.blueprint
+                }
+            ]
+        }
+        entities_at_port = await ocean.port_client.search_entities(
+                        UserAgentType.exporter,
+                        query
+                    )
```

Comment:
note that there are entities that can be ingested using userAgentType.gitops, see gitlab for example.

This can be deferred. so maybe what should be passed to here is the userAgent